### PR TITLE
perf: Optimize oracle Filter* functions

### PR DESCRIPTION
**Pull request by robert-zaremba**
## Description

Small optimization in Oracle package

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: price-feeder/oracle/filter.go [hunk 1]
```diff
@@ -122,14 +126,15 @@ func FilterCandleDeviations(
...
 			}
 
-			if _, ok := deviations[base]; !ok ||
-				(price.GTE(means[base].Sub(deviations[base].Mul(threshold))) &&
-					price.LTE(means[base].Add(deviations[base].Mul(threshold)))) {
-				if _, ok := filteredCandles[providerName]; !ok {
-					filteredCandles[providerName] = make(map[string][]provider.CandlePrice)
+			if d, ok := deviations[base]; !ok ||
+				(price.GTE(means[base].Sub(d.Mul(threshold))) &&
+					price.LTE(means[base].Add(d.Mul(threshold)))) {
```
Comment (RafilxTenfen) [inline price-feeder/oracle/filter.go:131]: I know it was not done in this PR, but this is getting confusing, could we move to a separate function?

File: price-feeder/oracle/filter.go [hunk 2]
```diff
@@ -25,11 +25,13 @@ func FilterTickerDeviations(
 	)
 
 	for providerName, priceTickers := range prices {
-		if _, ok := priceMap[providerName]; !ok {
-			priceMap[providerName] = make(map[string]sdk.Dec)
+		p, ok := priceMap[providerName]
```
[the remark this answers is not in the document]
Comment (robert-zaremba) [reply price-feeder/oracle/filter.go:28]: Isn't it clear from the context? It's only used inside this block, which is few lines long.
See the guideline: https://talks.golang.org/2014/names.slide#6

File: price-feeder/oracle/filter.go [hunk 3]
```diff
@@ -86,11 +89,12 @@ func FilterCandleDeviations(
 		candlePrices := make(provider.AggregatedProviderCandles)
 
 		for base, cp := range priceCandles {
-			if _, ok := candlePrices[providerName]; !ok {
-				candlePrices[providerName] = make(map[string][]provider.CandlePrice)
+			p, ok := candlePrices[providerName]
```
[the remark this answers is not in the document]
Comment (robert-zaremba) [reply price-feeder/oracle/filter.go:90]: which one we are reusing?

File: price-feeder/oracle/filter.go (hunk 1 above)
Comment (robert-zaremba) [reply price-feeder/oracle/filter.go]: The comparison (if condition)? added: 71dffcf8589590635c5d23ac5d22d3a949cb430b

**Review by RafilxTenfen — APPROVED** [review]

File: price-feeder/oracle/filter.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (robert-zaremba) [reply price-feeder/oracle/filter.go:90]: neh, it's already in the context. I guess `p` is more clear then `fp`

**Pull request merged**